src/index.ts

```diff
@@ -0,0 +1,76 @@
+import {IPC} from 'node-ipc';
+import {LanguageService} from 'typescript/lib/tsserverlibrary';
+
+import {SimpleDiagnostic} from './Protocol';
+
+function init(): ts.server.PluginModule {
+  function create(info: ts.server.PluginCreateInfo): LanguageService {
+    const ipc = new IPC();
+    ipc.config.id = 'ts-expose-status-plugin-server';
+    // We MUST silence logging, or else node-ipc will print to stdout. VSCode talks with the TS
+    // language service over stdin/stdout, so any extraneous messages to stdout will break VSCode's
+    // normal code intelligence features.
+    ipc.config.silent = true;
+
+    function log(s: string): void {
+      info.project.projectService.logger.info(s);
+    }
+
+    function convertToSimpleDiagnostic(diagnostic: ts.Diagnostic): SimpleDiagnostic {
+      return {
+        filePath: diagnostic.file ? diagnostic.file.fileName : null,
+        start: diagnostic.start != null ? diagnostic.start : null,
+        end:
+          diagnostic.start != null && diagnostic.length != null ? diagnostic.start + diagnostic.length : null,
+        message:
+          typeof diagnostic.messageText === 'string'
+            ? diagnostic.messageText
+            : diagnostic.messageText.messageText,
+        code: diagnostic.code,
+      };
+    }
+
+    function getAllErrors(): Array<SimpleDiagnostic> {
+      const program = info.languageService.getProgram();
+      if (!program) {
+        return [];
+      }
+      const tsDiagnostics = [...program.getSemanticDiagnostics(), ...program.getSyntacticDiagnostics()];
```

Comment:
Can you leave a comment about why we only care about these two? 

What about getConfigFileParsingDiagnostics, getOptionsDiagnostics, and getGlobalDiagnostics?

Reply:
👍 my understanding is that those other diagnostics are obscure, but TBH I didn't look into them a whole lot. Regardless, I think excluding occasional errors is fine because it all gets checked in CI in the long run anyway.